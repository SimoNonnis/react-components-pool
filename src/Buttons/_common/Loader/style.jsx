import styled from 'styled-components';
import Spinner from '../../../Icons/Spinner';

export const SpinnerContainer = styled.div`
  position: relative;
`;

export const SpinnerStyled = styled(Spinner)`
  position: absolute;
  margin: 0 auto;
  left: 0;
  right: 0;
  top: -3px;
  ${({ size }) => size === 'large' && 'top: -5px'};
  ${({ size }) => size === 'small' && 'top: 0'};
`;
