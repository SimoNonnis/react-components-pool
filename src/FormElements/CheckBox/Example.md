CheckBox with default props

```jsx
initialState = { checked: true };

<CheckBox
  id="yes"
  label="Yes"
  name="choice"
  checked={state.checked}
  onChange={() => setState({ checked: !state.checked })}
/>;
```

```jsx
initialState = { checked: true };

<CheckBox
  id="yes1"
  label="I am a very long label and i am hopeing that i will wrap to the next line as i need to test! I am a very long label and i am hopeing that i will wrap to the next line as i need to test!"
  name="choice"
  checked={state.checked}
  onChange={() => setState({ checked: !state.checked })}
/>;
```

Checkbox with label as styled component
```jsx
initialState = { checked: true };
const LabelComp = require('./style').ExampleLabel;
<CheckBox
  id="yes2"
  label={<LabelComp>Label</LabelComp>}
  name="choice"
  checked={state.checked}
  onChange={() => setState({ checked: !state.checked })}
/>;
```

Checkbox with label as HTML element
```jsx
initialState = { checked: true };
<CheckBox
  id="yes3"
  label={<div>Label</div>}
  name="choice"
  checked={state.checked}
  onChange={() => setState({ checked: !state.checked })}
/>;
```

Checkbox with label as function
```jsx
initialState = { checked: true };
const LabelComp = require('./style').ExampleLabel;
<CheckBox
  id="yes4"
  label={() => <div>Label</div>}
  name="choice"
  checked={state.checked}
  onChange={() => setState({ checked: !state.checked })}
/>;
```

CheckBox with error

```jsx
initialState = { checked: false };

<CheckBox
  id="iagree"
  label="I agree"
  name="choice1"
  error="error string"
  checked={state.checked}
  onChange={() => setState({ checked: !state.checked })}
/>;
```

Multiple CheckBoxes with default props

```jsx
initialState = { checked1: false, checked2: true, checked3: false, checked4: false };
<div>
  <CheckBox
    id="london"
    label="London"
    name="cities"
    checked={state.checked1}
    onChange={() => setState({ checked1: !state.checked1 })}
  />
  <CheckBox
    id="rome"
    label="Rome"
    name="cities"
    checked={state.checked2}
    onChange={() => setState({ checked2: !state.checked2 })}
  />
  <CheckBox
    id="madrid"
    label="Madrid"
    name="cities"
    checked={state.checked3}
    onChange={() => setState({ checked3: !state.checked3 })}
  />
  <CheckBox
    id="madrid"
    label="Madrid (disabled)"
    name="cities"
    disabled
    checked={state.checked4}
    onChange={() => setState({ checked3: !state.checked4 })}
  />
</div>;
```
