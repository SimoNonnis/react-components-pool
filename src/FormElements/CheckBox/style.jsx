import styled from 'styled-components';

export const Container = styled.span`
  display: inline-block;
  position: relative;

  & + & {
    margin-left: 1em;
  }
`;
Container.displayName = 'Container';

export const Check = styled.input`
  opacity: 0;
  position: absolute;
  right: 0;

  &:checked + label:after {
    opacity: 1;
    transform: rotate(45deg) scale(1);
  }
`;
Check.displayName = 'Check';

export const Label = styled.label`
  cursor: pointer;
  display: inline-block;
  font-family: ${({ theme }) => theme.base.fontFamily};
  font-size: 16px;
  color: ${({ theme }) => theme.color.veryDarkBlue};
  ${({ theme, error }) => error && `color: ${theme.color.red}`};
  ${({ theme, disabled }) => disabled && `color: ${theme.color.lightGrey1}`};
  line-height: 24px;
  position: relative;
  user-select: none;
  padding-left: 30px;

  &:before {
    content: '';
    position: absolute;
    left: 0;
    top: 0;
    width: 20px;
    height: 20px;
    border: 1px solid;
    border-color: ${({ theme: { color }, disabled }) =>
      disabled ? color.lightGrey1 : color.stoneGrey};
    background-color: ${({ theme }) => theme.color.white};
    border-radius: 3px;
    box-shadow: inset 0 1px 2px rgba(0, 0, 0, 0.1);
  }

  ${({ theme, disabled }) =>
    disabled ? `&:hover { &:before { background-color: ${theme.color.lightGrey3}; } }` : null};
  &:after {
    border: solid ${({ theme }) => theme.color.mediumBlue};
    border-width: 0 3px 3px 0;
    content: '';
    font-family: ${({ theme }) => theme.base.fontFamily};
    font-size: 16px;
    height: 10px;
    left: 7px;
    top: 2.5px;
    opacity: 0;
    position: absolute;
    transform: rotate(45deg) scale(0);
    transition: all 0.2s;
    width: 5px;
  }
`;
Label.displayName = 'Label';

export const ExampleLabel = styled.div`
  color: ${({ theme }) => theme.color.green};
`;
