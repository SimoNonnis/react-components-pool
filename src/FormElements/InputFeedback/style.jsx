import styled from 'styled-components';

export const ErrorStyled = styled.div`
  color: ${({ theme }) => theme.color.red};
  font-size: ${({ theme }) => theme.base.fontSize};
  margin-top: 7px;
  min-height: 17px;
`;
