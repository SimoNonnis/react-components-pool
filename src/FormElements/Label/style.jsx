import styled from 'styled-components';

export const LabelStyled = styled.label`
  color: ${({ theme }) => theme.color.veryDarkBlue};
  font-family: ${({ theme }) => theme.base.fontFamily};
  ${({ theme, error }) => error && `color: ${theme.color.red}`};
  ${({ theme, disabled }) => disabled && `color: ${theme.color.lightGrey2}`};
  margin-bottom: 7px;
  display: block;
  font-size: ${({ theme }) => theme.base.fontSize};
  font-weight: 600;
`;
LabelStyled.displayName = 'LabelStyled';

export const Asterisk = styled.span`
  color: ${({ theme, disabled }) => (disabled ? theme.color.lightGrey2 : theme.color.red)};
  display: inline-block;
  margin-left: 3px;
`;
Asterisk.displayName = 'Asterisk';
