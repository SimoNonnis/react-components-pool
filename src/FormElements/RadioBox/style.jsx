import styled from 'styled-components';

export const Container = styled.span`
  display: inline-block;
  position: relative;

  & + & {
    margin-left: 1em;
  }
`;
Container.displayName = 'Container';

export const Radio = styled.input`
  opacity: 0;
  position: absolute;
  right: 0;

  &:checked + label:after {
    opacity: 1;
    transform: scale(1);
  }
`;
Radio.displayName = 'Radio';

export const Label = styled.label`
  cursor: pointer;
  font-family: ${({ theme }) => theme.base.fontFamily};
  color: ${({ theme }) => theme.color.veryDarkBlue};
  ${({ theme, error }) => error && `color: ${theme.color.red}`};
  ${({ theme, disabled }) => disabled && `color: ${theme.color.lightGrey1}`};
  font-size: 16px;
  line-height: 24px;
  position: relative;
  user-select: none;
  padding-left: 30px;
  display: inline-block;
  font-weight: 600;

  &:before {
    box-sizing: content-box;
    background-color: ${({ theme }) => theme.color.white};
    border-radius: 100%;
    border: 1px solid;
    border-color: ${({ theme: { color }, disabled }) =>
      disabled ? color.lightGrey1 : color.stoneGrey};
    box-shadow: inset 0 1px 2px rgba(0, 0, 0, 0.1);
    content: '';
    height: 20px;
    left: 0;
    position: absolute;
    top: 0;
    width: 20px;
  }

  ${({ theme, disabled }) =>
    disabled ? `&:hover { &:before { background-color: ${theme.color.lightGrey3}; } }` : null};
  &:after {
    box-sizing: content-box;
    background-color: ${({ theme }) => theme.color.mediumBlue};
    border-radius: 100%;
    content: '';
    opacity: 0;
    position: absolute;
    transform: scale(0);
    transition: all 0.2s;
    height: 12px;
    width: 12px;
    top: 5px;
    left: 5px;
  }
`;
Label.displayName = 'Label';
