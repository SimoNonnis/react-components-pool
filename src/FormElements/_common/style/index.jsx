import styled, { css } from 'styled-components';

const basicStyle = css`
  box-shadow: none;
  box-sizing: border-box;
  display: block;
  font-family: ${({ theme }) => theme.base.fontFamily};
  font-size: 16px;
  font-weight: inherit;
  height: auto;
  line-height: 24px;
  margin: 0;
  width: 100%;
  &:focus {
    outline: none;
    border-color: ${({ theme }) => theme.color.mediumBlue};
  }
`;

const borderStyle = css`
  border-radius: 2px;
  border: 1px solid;
`;

const defaultColor = css`
  ${borderStyle};
  border-color: ${({ theme }) => theme.color.stoneGrey};
  color: ${({ theme }) => theme.color.veryDarkBlue};
  caret-color: ${({ theme }) => theme.color.veryDarkBlue};
  &:focus {
    border-color: ${({ theme }) => theme.color.mediumBlue};
  }
  &::-webkit-datetime-edit-year-field,
  &::-webkit-datetime-edit-day-field,
  &::-webkit-datetime-edit-month-field {
    color: ${({ theme }) => theme.color.veryDarkBlue};
    font-weight: inherit;
  }
`;

const errorColor = css`
  ${borderStyle};
  border-color: ${({ theme }) => theme.color.red};
  &:focus {
    border-color: ${({ theme }) => theme.color.red};
  }
`;

const disabledColor = css`
  color: ${({ theme }) => theme.color.lightGrey2};
  border-color: ${({ theme }) => theme.color.lightGrey2};
  caret-color: ${({ theme }) => theme.color.lightGrey2};
  &:focus {
    border-color: ${({ theme }) => theme.color.lightGrey2};
  }
  &::-webkit-datetime-edit-year-field,
  &::-webkit-datetime-edit-day-field,
  &::-webkit-datetime-edit-month-field {
    color: ${({ theme }) => theme.color.lightGrey2};
    font-weight: inherit;
  }
`;

export const InputStyled = styled.input`
  ${basicStyle};
  ${defaultColor};
  ${({ error }) => error && errorColor};
  ${({ disabled }) => disabled && disabledColor};
  padding: 12px 15px;
`;

export const SelectStyled = styled.select`
  ${basicStyle};
  color: ${({ theme, disabled }) => (disabled ? theme.color.lightGrey2 : theme.color.veryDarkBlue)};
  padding: 13px 50px 13px 15px;
  -ms-appearance: none !important;
  appearance: none !important;
  background-image: none !important;
  background: transparent !important;
  border-radius: 0 !important;
  border: 0 !important;
  box-shadow: none !important;
  outline: 0 !important;
`;

export const SelectWrap = styled.div`
  ${defaultColor};
  ${({ error }) => error && errorColor};
  ${({ disabled }) => disabled && disabledColor};
  position: relative;
  &:after {
    border-left: 6px solid transparent;
    border-right: 6px solid transparent;
    border-top: 8px solid
      ${({ theme, disabled }) => (disabled ? theme.color.lightGrey2 : theme.color.veryDarkBlue)};
    content: '';
    height: 0;
    pointer-events: none;
    position: absolute;
    right: 20px;
    top: 50%;
    transform: translateY(-50%);
    transition: all 0.2s cubic-bezier(0.25, 0.46, 0.45, 0.94);
    width: 0;
    z-index: 2;
  }
`;

export const TextareaStyled = styled.textarea`
  ${basicStyle};
  ${defaultColor};
  ${({ error }) => error && errorColor};
  ${({ disabled }) => disabled && disabledColor};
  padding: 15px;
  resize: ${({ disabled }) => (disabled ? 'none' : 'vertical')};
`;
