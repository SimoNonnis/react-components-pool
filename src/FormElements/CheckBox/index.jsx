import React from 'react';
import PropTypes from 'prop-types';

import { Container, Check, Label } from './style';

const propTypes = {
  id: PropTypes.string.isRequired,
  label: PropTypes.oneOfType([PropTypes.func, PropTypes.node, PropTypes.string]).isRequired,
  name: PropTypes.string.isRequired,
  className: PropTypes.string,
  checked: PropTypes.bool,
  error: PropTypes.string,
  disabled: PropTypes.bool,
};
const defaultProps = {
  className: null,
  checked: false,
  error: null,
  disabled: false,
};

const CheckBox = ({ className, id, name, label, error, checked, disabled, ...props }) => {
  const labelComp = typeof label === 'function' ? label() : label;
  return (
    <Container className={className}>
      <Check disabled={disabled} type="checkbox" id={id} name={name} checked={checked} {...props} />
      <Label disabled={disabled} htmlFor={id} error={!!error}>
        {labelComp}
      </Label>
    </Container>
  );
};

CheckBox.propTypes = propTypes;
CheckBox.defaultProps = defaultProps;

export default CheckBox;
