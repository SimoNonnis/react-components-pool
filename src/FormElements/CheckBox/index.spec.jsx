/* eslint-env jest */
import React from 'react';

import 'jest-styled-components';

import CheckBox from './';
import { Check, ExampleLabel } from './style';

describe('Test <CheckBox />', () => {
  it('should render CheckBox correctly with only required props', () => {
    const wrapper = shallow(<CheckBox id="yes" label="Yes" name="choice" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render CheckBox correctly if checked true', () => {
    const wrapper = shallow(<CheckBox id="yes" label="Yes" name="choice" checked />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render CheckBox correctly with a label as a string', () => {
    const wrapper = shallow(<CheckBox id="yes" label="Label" name="choice" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render CheckBox correctly component with label as a function', () => {
    const wrapper = shallow(
      <CheckBox id="yes" label={() => <span>Label component</span>} name="choice" />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render CheckBox correctly component with label as a styled component', () => {
    const labelComp = <ExampleLabel>Label</ExampleLabel>;
    const wrapper = shallow(<CheckBox id="yes" label={labelComp} name="choice" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should call handleChange when checkbox is clicked', () => {
    const handleChangeMock = jest.fn();
    const wrapper = shallow(
      <CheckBox id="yes" label="Yes" name="choice" checked onChange={handleChangeMock} />
    );
    wrapper.find(Check).simulate('change');
    expect(handleChangeMock).toHaveBeenCalled();
  });

  it('should call handleBlur when checkbox is clicked', () => {
    const handleChangeMock = jest.fn();
    const handleBlurMock = jest.fn();

    const wrapper = shallow(
      <CheckBox
        id="yes"
        label="Yes"
        name="choice"
        checked
        onChange={handleChangeMock}
        onBlur={handleBlurMock}
      />
    );
    wrapper.find(Check).simulate('blur');
    expect(handleBlurMock).toHaveBeenCalled();
  });
});
