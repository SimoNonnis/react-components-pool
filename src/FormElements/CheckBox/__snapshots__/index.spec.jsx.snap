// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <CheckBox /> should render CheckBox correctly component with label as a function 1`] = `
<Container
  className={null}
>
  <Check
    checked={false}
    disabled={false}
    id="yes"
    name="choice"
    type="checkbox"
  />
  <Label
    disabled={false}
    error={false}
    htmlFor="yes"
  >
    <span>
      Label component
    </span>
  </Label>
</Container>
`;

exports[`Test <CheckBox /> should render CheckBox correctly component with label as a styled component 1`] = `
<Container
  className={null}
>
  <Check
    checked={false}
    disabled={false}
    id="yes"
    name="choice"
    type="checkbox"
  />
  <Label
    disabled={false}
    error={false}
    htmlFor="yes"
  >
    <styled.div>
      Label
    </styled.div>
  </Label>
</Container>
`;

exports[`Test <CheckBox /> should render CheckBox correctly if checked true 1`] = `
<Container
  className={null}
>
  <Check
    checked={true}
    disabled={false}
    id="yes"
    name="choice"
    type="checkbox"
  />
  <Label
    disabled={false}
    error={false}
    htmlFor="yes"
  >
    Yes
  </Label>
</Container>
`;

exports[`Test <CheckBox /> should render CheckBox correctly with a label as a string 1`] = `
<Container
  className={null}
>
  <Check
    checked={false}
    disabled={false}
    id="yes"
    name="choice"
    type="checkbox"
  />
  <Label
    disabled={false}
    error={false}
    htmlFor="yes"
  >
    Label
  </Label>
</Container>
`;

exports[`Test <CheckBox /> should render CheckBox correctly with only required props 1`] = `
<Container
  className={null}
>
  <Check
    checked={false}
    disabled={false}
    id="yes"
    name="choice"
    type="checkbox"
  />
  <Label
    disabled={false}
    error={false}
    htmlFor="yes"
  >
    Yes
  </Label>
</Container>
`;
